bin/validate.py

```diff
@@ -300,22 +300,25 @@ def run(self, testcase, mode, constraints=None, args=None):
             path = testcase.out_path.resolve()
         else:
             assert mode != Mode.INPUT
+            # mode is actually a run
             path = mode.out_path
+            in_path = mode.in_path
 
         if self.language in Validator.FORMAT_VALIDATOR_LANGUAGES:
             raise ValueError("Invalid output validator language")
 
         cwd, constraints_path, arglist = self._run_helper(testcase, constraints, args)
-        feedbackdir = cwd if isinstance(mode, Mode) else mode.feedbackdir
+        if not isinstance(mode, Mode):
```

Comment:
```suggestion
        if isinstance(mode, run.Run):
```

Reply:
hm, will undo this again... cyclic imports...